graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/AggregationTabularResponseCreator.java

```diff
@@ -40,31 +41,43 @@ public class AggregationTabularResponseCreator implements TabularResponseCreator
 
     public TabularResponse mapToResponse(final AggregationRequestSpec searchRequestSpec, final SearchJob searchJob) throws AggregationFailedException {
         final SearchJobDTO searchJobDTO = SearchJobDTO.fromSearchJob(searchJob);
+
+
         final QueryResult queryResult = searchJobDTO.results().get(SearchRequestSpecToSearchMapper.QUERY_ID);
 
         if (queryResult != null) {
             throwErrorIfAnyAvailable(queryResult);
             final SearchType.Result aggregationResult = queryResult.searchTypes().get(AggregationSpecToPivotMapper.PIVOT_ID);
+
             if (aggregationResult instanceof PivotResult pivotResult) {
-                return mapToResponse(searchRequestSpec, pivotResult);
+                final List<SeriesSpec> requestedSeries = extractSeries(queryResult);
```

Comment:
@luk-kaminski this is the most important change - I am obtaining the actual series (and their literals) here, so I can use them to extract metric values from pivRow.values later.